src/safeposix/syscalls/net_calls.rs

```diff
@@ -829,10 +1023,15 @@ impl Cage {
             };
         }
 
+        //Setup the socket handle as connected, insert the remote address of the connection,
+        //and reset the errno in case it is set to EINPROGRESS
         sockhandle.state = ConnState::CONNECTED;
         sockhandle.remoteaddr = Some(remoteaddr.clone());
         sockhandle.errno = 0;
         // set the rawfd for select
+        // ** What does the above comment mean ?? ** //
+        //The raw fd of the socket is the set to be the same as the fd set by the kernal in the libc connect call
+        // ** Will this ever cause issues of indexing into an fd that is already set by lind ?? ** //
```

Comment:
A AF_INET/AF_INET6 socket will have two fds, one fd is created by lind itself and integrated into its own fd system, another fd is rawfd, which is the real file descriptor of the socket used by the actual operating system. Since in select_syscall, we cannot implement the select logics for AF_INET socket right now, so we have to call the actual `select` syscall from libc, which takes the rawfd as the argument instead of the fake fd used by lind.